MimeKit/Utils/MimeUtils.cs

```diff
@@ -116,10 +115,7 @@ public static string GenerateMessageId (string domain)
 		/// <returns>The message identifier.</returns>
 		public static string GenerateMessageId ()
 		{
-#if NETSTANDARD1_3 || NETSTANDARD1_6
-			return GenerateMessageId ("localhost.localdomain");
-#else
-			if (DefaultHostName == null) {
+			if (DefaultHostName is null) {
 				try {
```

Comment:
What are the implications of `is null` vs `== null`?

Reply:
== null will use the type's == operator (when defined), while is null will check if the underlying object is null. Is null has a nice benefit of ensuring an equity operator doesn't accidentally change the behavior -- and can be generate better machine code in some cases.